### PR TITLE
Updates charms using `orc8r_base_db`

### PR DESCRIPTION
**Pull request by sanchezfdezjavier**
# Description

Uses `orc8r_base_db` changes from this [PR](https://github.com/canonical/magma-orc8r-libs/pull/11) for all the charms using the library.  We also make the corresponding changes in the bundle.

- orc8r-accessd
- orc8r-configurator
- orc8r-ctraced
- orc8r-device
- orc8r-directoryd
- orc8r-lte
- orc8r-policydb
- orc8r-smsd
- orc8r-state
- orc8r-subscriberdb
- orc8r-subscriberdb-cache
- orc8r-tenants

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [ ] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have bumped the version of any required library.

### CODE REVIEW

**Comment by sanchezfdezjavier** [comment]
Thanks for reviewing this huge PR, if done separately would take ages!

File: orc8r-bundle/bundle.yaml.j2 [hunk 1]
```diff
@@ -366,8 +366,8 @@ relations:
   - orc8r-certifier
 - - nms-nginx-proxy:magma-nms-magmalte
   - nms-magmalte:magma-nms-magmalte
-- - orc8r-accessd:db
-  - postgresql-k8s:db
+- - orc8r-accessd:database
```
Comment (ghislainbourgeois) [inline orc8r-bundle/bundle.yaml.j2:369]: Our side of all the relations should stay `db`, as that is how the library finds the relation.

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: orc8r-bundle/bundle.yaml.j2 (hunk 1 above)
Comment (sanchezfdezjavier) [reply orc8r-bundle/bundle.yaml.j2:369]: Fixed.

File: orc8r-accessd-operator/metadata.yaml [hunk 2]
```diff
@@ -26,5 +25,5 @@ provides:
 
 requires:
   db:
```
Comment (gruyaume) [inline orc8r-accessd-operator/metadata.yaml]: Aren't we now using `database` as the relation name?

Comment (sanchezfdezjavier) [reply orc8r-accessd-operator/metadata.yaml]: Yes, I was reverting the change. Now it's good.

**Review by ghislainbourgeois — APPROVED** [review]
LGTM, but don't forget to change the relation names in the integration tests as you merge other PRs. (certifier and bootstrapper specifically)

**Review by Gmerold — APPROVED** [review]

**Review by saltiyazan — APPROVED** [review]

**Review by gruyaume — APPROVED** [review]
Approved (assuming CI passes)

**Comment by sanchezfdezjavier** [comment]
> Approved (assuming CI passes)

Currently, we're blocked by this [postgres bug](https://warthogs.atlassian.net/browse/DPE-1470). itests fail when we remove postgres and redeploy it, it gets stuck in `awaiting cluster start`. I've talked to Marcelo, he's struggling to reproduce the issue, the only fix found was to patch the k8s endpoints, which I think is outside the scope of these PRs.

We've discussed in the daily setting the affected tests to `xfail` until the issue is fixed. Besides, `main-branch` tests are passing because we're not testing this behavior, just itests for individual charms are failing.

Let me know if that works and I push the corresponding changes.

**Comment by ghislainbourgeois** [comment]
> > Approved (assuming CI passes)
> 
> Currently, we're blocked by this [postgres bug](https://warthogs.atlassian.net/browse/DPE-1470). itests fail when we remove postgres and redeploy it, it gets stuck in `awaiting cluster start`. I've talked to Marcelo, he's struggling to reproduce the issue, the only fix found was to patch the k8s endpoints, which I think is outside the scope of these PRs.
> 
> We've discussed in the daily setting the affected tests to `xfail` until the issue is fixed. Besides, `main-branch` tests are passing because we're not testing this behavior, just itests for individual charms are failing.
> 
> Let me know if that works and I push the corresponding changes.

I agree with setting the tests to `xfail`. On my side, I was able to reproduce the issue by doing `juju remove-application postgresql-k8s --force --no-wait`. I am not sure those particular steps are supposed to be working however, as they probably don't let the charm run its cleanup.

**Pull request merged**